### ISSUE TITLE
Diligent libraries compiled with Intel does not link when compile on AMD

### ISSUE THREAD

**Issue by chelvisgit** [DiligentGraphics/DiligentEngine#323]
Trying to link our project with previously compiled Diligent lib/dlls (on Intel), we are receiving this message when we try to compile it with AMD 7950x:

**Error C1047 The object or library file '..\..\lib\Release\Diligent-Win32Platform.lib' was created by a different version of the compiler than other objects like '..\..\lib\xxxxxx.lib'; rebuild all objects and libraries with the same compiler**

Machine A:
OS: Windows 11 Pro 22H2
Build tools: Visual Studio 2022 14.41.34120
CPU: 13th Gen Intel(R) Core(TM) i7-13800H 2.50 GHz
GPU: NVIDIA GeForce RTX 4080 Laptop GPU
Driver: NVIDIA 565.90

Machine B:
Build tools: Visual Studio 2022 14.41.34120
CPU: AMD 7950x

**Comment by TheMostDiligent** [comment]
I am not sure what the problem here is.
Maybe Visual Studio uses different compiler settings on these machines.

**Comment by chelvisgit** [comment]
Trying to link the libraries with AMD (compiled with intel CPU) we also receive this error:
LINK : fatal error C1007: unrecognized flag '-archSSE2' in 'p2'

**Comment by TheMostDiligent** [comment]
SSE2 should be fully supported by AMD

**Comment by AnloWho** [comment]
I agree this looks like that the same compiler version isn't used for compile & link.
